// multion/dist/serialization/client/requests/RetrieveInput.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RetrieveInput = void 0;
const core = __importStar(require("../../../core"));
const Format_1 = require("../../types/Format");
exports.RetrieveInput = core.serialization.object({
    cmd: core.serialization.string(),
    url: core.serialization.string().optional(),
    sessionId: core.serialization.property("session_id", core.serialization.string().optional()),
    local: core.serialization.boolean().optional(),
    fields: core.serialization.list(core.serialization.string()).optional(),
    format: Format_1.Format.optional(),
    maxItems: core.serialization.property("max_items", core.serialization.number().optional()),
    fullPage: core.serialization.property("full_page", core.serialization.boolean().optional()),
    renderJs: core.serialization.property("render_js", core.serialization.boolean().optional()),
    scrollToBottom: core.serialization.property("scroll_to_bottom", core.serialization.boolean().optional()),
    includeScreenshot: core.serialization.property("include_screenshot", core.serialization.boolean().optional()),
});
